besu/src/test/java/org/hyperledger/besu/cli/BesuCommandTest.java

```diff
@@ -5441,6 +5442,64 @@ public void portInUseReportsError() throws IOException {
     serverSocket.close();
   }
 
+  @Test
+  public void checkpointPostMergeShouldFailWhenGenesisHasNoTTD() throws IOException {
+    final String configText =
+        Resources.toString(
+            Resources.getResource("invalid_post_merge_near_head_checkpoint.json"),
+            StandardCharsets.UTF_8);
+    final Path genesisFile = createFakeGenesisFile(new JsonObject(configText));
+
+    parseCommand(
+        "--genesis-file",
+        genesisFile.toString(),
+        "--sync-mode",
+        "X_CHECKPOINT",
+        "--Xcheckpoint-post-merge-enabled");
+
+    assertThat(commandOutput.toString(UTF_8)).isEmpty();
+    assertThat(commandErrorOutput.toString(UTF_8))
+        .contains("Near head checkpoint sync requires TTD in the genesis file");
+  }
+
+  @Test
+  public void checkpointPostMergeShouldFailWhenGenesisUsesCheckpointFromPreMerge() {
+    // using the default genesis which has a checkpoint sync block prior to the merge
+    parseCommand("--sync-mode", "X_CHECKPOINT", "--Xcheckpoint-post-merge-enabled");
+
+    assertThat(commandOutput.toString(UTF_8)).isEmpty();
+    assertThat(commandErrorOutput.toString(UTF_8))
+        .contains(
+            "Near head checkpoint sync requires a block with total difficulty greater than the TTD");
+  }
+
+  @Test
+  public void checkpointPostMergeShouldFailWhenSyncModeIsNotCheckpoint() {
+
+    parseCommand("--sync-mode", "X_SNAP", "--Xcheckpoint-post-merge-enabled");
+    assertThat(commandErrorOutput.toString(UTF_8))
+        .contains("--Xcheckpoint-post-merge-enabled can only be used with X_CHECKPOINT sync-mode");
+  }
+
+  @Test
+  public void checkpointPostMergeWithPostMergeBlockSucceeds() throws IOException {
+    final String configText =
+        Resources.toString(
+            Resources.getResource("valid_post_merge_near_head_checkpoint.json"),
+            StandardCharsets.UTF_8);
+    final Path genesisFile = createFakeGenesisFile(new JsonObject(configText));
+
+    parseCommand(
+        "--genesis-file",
+        genesisFile.toString(),
+        "--sync-mode",
+        "X_CHECKPOINT",
+        "--Xcheckpoint-post-merge-enabled");
+
+    assertThat(commandOutput.toString(UTF_8)).isEmpty();
+    assertThat(commandErrorOutput.toString(UTF_8)).isEmpty();
+  }
+
```

Comment:
Tests for TTD < 0, TTD = 0, TTD > 0 would be good. Maybe in its own class if that means you can avoid creating load of json files?

Reply:
TTD < 0? What case would we be testing here?

Reply:
sorry I mistyped, I meant TD < TTD (pre-merge), TTD = TD, TD > TTD (post-merge) and finally the TD = 0 special case

Reply:
Added some tests in the BesuController to make sure we're using the right controller in each case.